appassembler-maven-plugin/src/main/java/org/codehaus/mojo/appassembler/daemon/yajsw/YetAnotherJavaServiceWrapperDaemonGenerator.java

```diff
@@ -0,0 +1,317 @@
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.codehaus.mojo.appassembler.daemon.yajsw;
+
+import org.apache.commons.io.FileUtils;
+import org.apache.maven.artifact.Artifact;
+import org.apache.maven.artifact.repository.layout.ArtifactRepositoryLayout;
+import org.apache.maven.project.MavenProject;
+import org.codehaus.mojo.appassembler.daemon.DaemonGenerationRequest;
+import org.codehaus.mojo.appassembler.daemon.DaemonGenerator;
+import org.codehaus.mojo.appassembler.daemon.DaemonGeneratorException;
+import org.codehaus.mojo.appassembler.model.Daemon;
+import org.codehaus.mojo.appassembler.util.DependencyFactory;
+import org.codehaus.mojo.appassembler.util.FormattedProperties;
+import org.codehaus.plexus.logging.AbstractLogEnabled;
+import org.codehaus.plexus.util.IOUtil;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.Reader;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Properties;
+
+/**
+ * @author Sandra Parsick
+ * @version $Id$
+ * @plexus.component role-hint="yajsw"
+ */
+public class YetAnotherJavaServiceWrapperDaemonGenerator
+    extends AbstractLogEnabled
+    implements DaemonGenerator
+{
+
+    private static List<String> CORE_LIBS = new ArrayList<String>()
+    {
+        {
+            add("commons/commons-cli-1.4.jar");
+            add("commons/commons-collections-3.2.2.jar");
+            add("commons/commons-configuration2-2.3.jar");
+            add("commons/commons-io-2.6.jar");
+            add("commons/commons-lang3-3.8.1.jar");
+            add("commons/commons-lang-2.6.jar");
+            add("commons/commons-logging-1.1.jar");
+            add("commons/commons-text-1.6.jar");
+            add("commons/commons-vfs2-2.2.jar");
+            add("jna/jna-5.3.1.jar");
+            add("jna/jna-platform-5.3.1.jar");
+            add("netty/netty-all-4.1.36.Final.jar");
+            add("yajsw/ahessian.jar");
+        }
+    };
+
+    private static List<String> EXTENDED_LIBS = new ArrayList<String>()
+    {
+        {
+            add("velocity/velocity-1.7.jar");
+            add("yajsw/hessian4.jar");
+            add("groovy/groovy-2.5.7.jar");
+            add("groovy/groovy-patch.jar");
```

Comment:
Do all these libraries have to have hardocded versions?

Reply:
Unfortunately, yes, because the version number is part of their name in the path. These libs are part of the YAJSW jar, that I downloaded before. YAJSW needs these libs for working correctly. See also my comment in the [issue](https://github.com/mojohaus/appassembler/issues/42#issuecomment-230249341) about the structure of YAJSW

Reply:
I get what you're saying, but each time there is an upgrade of the `yajsw` library, these lines (here and below) will have to be manually updated. Would it be possible to implement a cleaner way of doing this? For example, you can scan the directories for these jar files by a pattern and resolve the paths this way. I know it's a bit of extra work, but it will save a lot of work in the future when there's upgrades (and you know how error prone these manual "touches" tend to be)...

Reply:
I completely with you. Maintenance will be very hard with this solution. I would spend more time for a better solution when one of maintainers give feedback whether I run in the right direction with this PR. 

IMHO the best solution would be when YAJSW would provide a normal Maven artifact in Maven central.

Reply:
> IMHO the best solution would be when YAJSW would provide a normal Maven artifact in Maven central.

Do they have an issue open for this that would could link to, (or make some noise)?

Reply:
@khmarbaise ,

Any feedback on this pull request?

Reply:
> > IMHO the best solution would be when YAJSW would provide a normal Maven artifact in Maven central.
> 
> Do they have an issue open for this that would could link to, (or make some noise)?

It exists an elder [ticket](https://sourceforge.net/p/yajsw/feature-requests/24/), but it doesn't address our problem.

@carlspring could you open a ticket and link it in our original issue?

Reply:
Not sure what you mean. Perhaps it would be easier/clearer for you to do it?